File: android2_lesson01/src/java/java/com/example/ama/android2_lesson01/db/base/BaseProviderDataManager.java
package com.example.ama.android2_lesson01.db.base;

import android.content.ContentUris;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.net.Uri;

import com.example.ama.android2_lesson01.db.provider.NotesDatabaseContract;

/**
 * Base implementation for {@link ProviderDataManager}
 * for work with {@link com.example.ama.android2_lesson01.db.provider.NotesProvider}
 */

public abstract class BaseProviderDataManager implements ProviderDataManager {

    protected Context context;

    public BaseProviderDataManager(Context context) {
        this.context = context;
    }

    @Override
    public Cursor query() {
        return context.getContentResolver()
                .query(NotesDatabaseContract.CONTENT_URI, null, null, null, null);
    }

    @Override
    public Uri insert(ContentValues values) {
        return context.getContentResolver().insert(NotesDatabaseContract.CONTENT_URI, values);
    }

    @Override
    public int update(long id, ContentValues values) {
        Uri uri = ContentUris.withAppendedId(NotesDatabaseContract.CONTENT_URI_ITEM, id);
        return context.getContentResolver().update(uri, values, null, null);
    }

    @Override
    public int delete(long id) {
        Uri uri = ContentUris.withAppendedId(NotesDatabaseContract.CONTENT_URI_ITEM, id);
        return context.getContentResolver().delete(uri, null, null);
    }
}

File: android2_lesson01/src/kotlin/java/com/example/ama/android2_lesson01/db/base/BaseProviderDataManager.kt
package com.example.ama.android2_lesson01.db.base

import android.content.ContentUris
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.net.Uri
import com.example.ama.android2_lesson01.db.provider.NotesProvider

/**
 * Base implementation for [ProviderDataManager]
 * for work with [NotesProvider]
 */

abstract class BaseProviderDataManager(protected val context: Context) : ProviderDataManager {

    override fun query(): Cursor =
            context.contentResolver
                    .query(NotesProvider.CONTENT_URI, null, null, null, null)


    override fun insert(values: ContentValues): Uri =
            context.contentResolver.insert(NotesProvider.CONTENT_URI, values)

    override fun update(id: Long, values: ContentValues): Int {
        val uri = ContentUris.withAppendedId(NotesProvider.CONTENT_URI_ITEM, id)
        return context.contentResolver.update(uri, values, null, null)
    }

    override fun delete(id: Long): Int {
        val uri = ContentUris.withAppendedId(NotesProvider.CONTENT_URI_ITEM, id)
        return context.contentResolver.delete(uri, null, null)
    }
}